packages/typescript-estree/src/convert.ts

```diff
@@ -2485,6 +2485,7 @@ export class Converter {
         const result = this.createNode<TSESTree.TSEnumMember>(node, {
           type: AST_NODE_TYPES.TSEnumMember,
           id: this.convertChild(node.name),
+          computed: node.name.kind === ts.SyntaxKind.ComputedPropertyName,
```

Comment:
can you set this value to undefined instead of false?

> Computed property names are not allowed in enums.

```ts
if (node.name.kind === ts.SyntaxKind.ComputedPropertyName) {
 computed: true
}
```

properties that comes from invalid syntax should be always set as optional

Reply:
> properties that comes from invalid syntax should be always set as optional

It doesn't just come from invalid syntax - the majority of time it is:

```ts
// semantically valid
enum Foo {
  ['a'],
  ['b'],
  ['c'],
}

// semantically invalid
const x = 'a', y = 'b', z = 'c';
enum Bar {
  [x],
  [y],
  [z],
}
```

Babel does not support the former case, even though it is semantically valid code. I haven't ever seen anyone do this, and prettier automatically replaces it with
````ts
enum Foo {
  'a',
  'b',
  'c',
}
```
[repl](https://prettier.io/playground/#N4Igxg9gdgLgprEAuECCuBbABAMQhLYAHSiywG0ByAQ0oF0AaEsqgI3qdIsrA5IF8QDEBAAOMAJbQAzslDUATgogB3AAqKEslNQBuECQBMhIVgupgA1nBgBlURYlQA5shgK0cYQAsYGADYA6t4S8NIOYHC2WqESuqEAnsjg0rLCTtJwCjBq5s4Y1MgAZtT+mcIAVtIAHgBC5lY2ttQYcAAyTnDFpeUgVdW2Ts7+cACKaBDw3WVeIA4KmQrJMAmicNJgChLiJqJbsIFGMN7IABwADMJ7EJmB5qLJe+tZul3CAI4T8Lli2iDU0gAtFA4HBDGCTAo4J8JFDctR8oUkCUZsJMhgJG4PLNpEMRuNJl1kT1ZjBqKxDoZjsgAEzCdzUCT+IYAYQgGAKySg0DeIDQmQAKuTtCjMvx+EA)

Reply:
> properties that comes from invalid syntax should be always set as optional

It doesn't just come from invalid syntax - the majority of time it is:

```ts
// semantically valid
enum Foo {
  ['a'],
  ['b'],
  ['c'],
}

// semantically invalid
const x = 'a', y = 'b', z = 'c';
enum Bar {
  [x],
  [y],
  [z],
}
```

Babel does not support the former case, even though it is semantically valid code. I haven't ever seen anyone do this, and prettier automatically replaces it with
````ts
enum Foo {
  'a',
  'b',
  'c',
}
```
[repl](https://prettier.io/playground/#N4Igxg9gdgLgprEAuECCuBbABAMQhLYAHSiywG0ByAQ0oF0AaEsqgI3qdIsrA5IF8QDEBAAOMAJbQAzslDUATgogB3AAqKEslNQBuECQBMhIVgupgA1nBgBlURYlQA5shgK0cYQAsYGADYA6t4S8NIOYHC2WqESuqEAnsjg0rLCTtJwCjBq5s4Y1MgAZtT+mcIAVtIAHgBC5lY2ttQYcAAyTnDFpeUgVdW2Ts7+cACKaBDw3WVeIA4KmQrJMAmicNJgChLiJqJbsIFGMN7IABwADMJ7EJmB5qLJe+tZul3CAI4T8Lli2iDU0gAtFA4HBDGCTAo4J8JFDctR8oUkCUZsJMhgJG4PLNpEMRuNJl1kT1ZjBqKxDoZjsgAEzCdzUCT+IYAYQgGAKySg0DeIDQmQAKuTtCjMvx+EA)